vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/ModuleManager.java

```diff
@@ -67,8 +68,12 @@ public IModule getModule(String id) {
     }
 
     @Override
-    public List<ISlide> getModuleSlides(String moduleId) {
-        return new ArrayList<>(slideRepo.findSlidesForModule(moduleId));
+    public HashMap<ISlide, Integer> getModuleSlides(String moduleId) {
```

Comment:
is that used anywhere?

Reply:
Yes it is used

Reply:
this is pretty ugly and avoidable (see my comment below). Also, I believe it is used in the sequence controller, which I don't think you have adapted.

Reply:
where? it's not used in any of your changes.